### ISSUE TITLE
Changes to package.json during prepublish are not published

### ISSUE THREAD

**Issue by spautz** [wclr/yalc#157]
I hit this issue after upgrading to Husky 5, using Pinst to disable my `postinstall` script before publishing (and restore it after publishing). It looks like Yalc publishes the original package.json to its store (with the 'wrong' `postinstall` intact)

I've confirmed that a full `npm publish` publishes the modified package.json.

### My setup

My library's package.json includes the `husky` and `pinst` commands suggested by the [Husky 5 docs](https://typicode.github.io/husky/#/?id=install):
```
    "postinstall": "husky install",
    "prepublishOnly": "yarn run build && yarn run pinst --disable",
    "postpublish": "pinst --enable",
```

`pinst` just rewrites package.json: the `postinstall` key gets renamed to `_postinstall` during publish (and renamed back after publish), so that `husky install` command ONLY runs when working with the package locally.

### The behavior I'm seeing

With the above, my original package.json get published to Yalc -- and then when I `yalc add` my package to a test project it fails because it tries to run `husky install`.

### Manual workaround

If I run `yarn pinst --disable` manually before `yalc publish`, then the updated package.json (with `"postinstall": "husky install"` renamed to `"_postinstall": ...`) is published.

The package then works fine when testing locally. I just run `yarn pinst --disable && yalc publish` whenever I want to publish a new version.

### What's happening inside Yalc

Starting at `publishPackage` [in publish.ts](https://github.com/wclr/yalc/blob/849f8ccac2b4b793bf01f9470444a22119161b18/src/publish.ts#L38):
1. Read and validate package.json (saved as `pkg`)
2. All the preScripts are run -- including `prepublishOnly`. My package.json gets modified, but the `pkg` var still holds its original content.
3. `copyPackageToStore` ([in copy.ts](https://github.com/wclr/yalc/blob/fdfc4d768117d0b3c247e537a7804b2d979a72ad/src/copy.ts#L113)) uses the original `pkg` content when writing the new package.json to the Yalc store ([line 189](https://github.com/wclr/yalc/blob/fdfc4d768117d0b3c247e537a7804b2d979a72ad/src/copy.ts#L189-L194)), without realizing that package.json was altered in step 2

### Possible solution

It looks like re-reading package.json to `pkg` after running preScripts (right before `copyPackageToStore`) might give the same behavior as npm, without any other side effects or changes in behavior.

That change worked well for me when testing locally. I'm happy to add a test and submit a PR if you'd like.

**Comment by wclr** [comment]
Ok, I think we can re-read the package manifest after pre scripts, will add this in the next version.

**Comment by silvenon** [comment]
I'm using `1.0.0-pre.53`, which contains 64f580e, but unfortunately it didn't solve the problem. 😕

**Comment by wclr** [comment]
> I'm using 1.0.0-pre.53, which contains 64f580e, but unfortunately it didn't solve the problem. 😕

Why?

**Comment by silvenon** [comment]
No, I assumed it wasn't specific to my project, but I was wrong, I tried it in a blank project and it worked. 👌 Then I tried `1.0.0-pre.50` just to be sure, and it published `postinstall` instead of `_postinstall` as expected.
